### PR TITLE
Tidy gpu CI workflow

### PR DESCRIPTION
**Pull request by rhodrin**
- Removes repeated example runs
- Examples are now tested with and without MPI
- All examples tested on openacc + openmp

### CODE REVIEW

**Comment by rhodrin** [comment]
NOTE: After investigating the opemp fails need to remember to re-add codecov for examples.

File: .github/workflows/pytest-gpu.yml [hunk 1]
```diff
@@ -66,31 +66,21 @@ jobs:
...
-    - name: Run examples
-      if: matrix.name == 'pytest-gpu-acc'
+    - name: Test examples
       run: |
-        python3 examples/seismic/elastic/elastic_example.py
-        python3 examples/seismic/elastic/elastic_example.py
-        python3 examples/seismic/tti/tti_example.py
-        python3 examples/seismic/viscoacoustic/viscoacoustic_example.py
-        python3 examples/seismic/viscoelastic/viscoelastic_example.py
+        pytest examples/seismic/elastic/elastic_example.py
```
Comment (FabioLuporini) [inline .github/workflows/pytest-gpu.yml:72]: should we keep the --cov etc ?

Comment (rhodrin) [reply .github/workflows/pytest-gpu.yml:72]: Yes, we should - see note above (in the PR that is, not in the yml).

**Comment by rhodrin** [comment]
@italoaug When you get the chance, would you mind taking a look at what's going on here? Thanks.

File: .github/workflows/pytest-gpu.yml [hunk 2]
```diff
@@ -66,31 +66,21 @@ jobs:
...
+        pytest examples/seismic/elastic/elastic_example.py
```
Comment (italoaug) [inline .github/workflows/pytest-gpu.yml:73]: The elastic example is tested twice without MPI.
The acoustic example is not tested with and without MPI.

**Review by FabioLuporini — APPROVED** [review]

**Pull request merged**